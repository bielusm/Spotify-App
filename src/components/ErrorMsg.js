import React from "react";
import { connect } from "react-redux";
import { Typography } from "@material-ui/core";

export const ErrorMsg = props => (
  <>
    {props.errMsg && (
      <Typography variant="h2" id="errMsg" className="my-4">
        {props.errMsg}
      </Typography>
    )}
  </>
);

const mapStateToProps = state => ({ errMsg: state.error });

export default connect(mapStateToProps)(ErrorMsg);
