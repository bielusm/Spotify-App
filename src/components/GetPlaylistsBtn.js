import React from "react";

const GetPlaylistsBtn = props => (
  <button
    id="getPlaylistsBtn"
    className="button"
    onClick={props.getPlaylists}
    disabled={props.getPlaylistsDisabled}
  >
    Pick playlists
  </button>
);

export default GetPlaylistsBtn;
