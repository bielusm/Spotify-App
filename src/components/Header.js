import React from "react";

const Header = () => <h1>Spotify Manager</h1>;
export default Header;
