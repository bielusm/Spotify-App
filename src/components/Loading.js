import React from "react";

export const Loading = props => (
  <div>
    {props.loading && (
      <div className="loadingStatus">
        <div className="spinner-border" role="status">
          <span className="sr-only"></span>
        </div>
        <h2 className="mx-2 d-inline">Loading...</h2>
      </div>
    )}
  </div>
);

export default Loading;
