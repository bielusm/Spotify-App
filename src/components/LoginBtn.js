import React from "react";

export default class LoginBtn extends React.Component {
  login = () => {
    const scope =
      "playlist-modify-public%20user-read-playback-state%20playlist-read-private%20playlist-modify-private";
    const redirect_uri = window.location.origin;
    //redirect_uri = redirect_uri.slice(0, redirect_uri.length - 1);
    const url = `https://accounts.spotify.com/authorize?client_id=4252feb807d04ced962e15f346258957&response_type=token&redirect_uri=${redirect_uri}&scope=${scope}&show_dialog=false`;
    window.location.href = url;
  };

  render() {
    return (
      <div>
        {this.props.loginVisible && (
          <button id="login" className="btn btn-secondary" onClick={this.login}>
            Log in
          </button>
        )}
      </div>
    );
  }
}
