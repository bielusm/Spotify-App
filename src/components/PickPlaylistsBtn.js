import React from "react";
import Button from "@material-ui/core/Button";

export const PickPlaylistsBtn = props => (
  <Button
    variant="contained"
    color="primary"
    id="pickPlaylistsBtn"
    onClick={props.getPlaylists}
    disabled={props.getPlaylistsDisabled}
  >
    Pick playlists
  </Button>
);

export default PickPlaylistsBtn;
