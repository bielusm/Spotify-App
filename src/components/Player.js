import React from "react";
import { Button, IconButton } from "@material-ui/core";

import {
  modifyLibrary,
  startPlayer,
  pausePlayer,
  nextTrack,
  prevTrack
} from "../api/player";

import { setErrorMsg } from "../store/actions/error";
import { connect } from "react-redux";

export class Player extends React.Component {
  togglePause = async () => {
    if (this.props.is_playing)
      pausePlayer(this.props.bearerToken).then(() => this.props.update());
    else startPlayer(this.props.bearerToken).then(() => this.props.update());
  };

  nextTrack = async () => {
    nextTrack(this.props.bearerToken).then(() => this.props.update());
  };

  prevTrack = async () => {
    prevTrack(this.props.bearerToken).then(() => {
      this.props.update();
    });
  };

  toggleLike = async () => {
    modifyLibrary(
      this.props.bearerToken,
      this.props.id,
      this.props.likes_song
    ).then(() => this.props.update());
  };
  render() {
    return (
      <>
        <IconButton
          className="prevBtn fas fa-backward"
          onClick={this.prevTrack}
        />

        <IconButton
          className={
            "playBtn fas " + (this.props.is_playing ? "fa-pause" : "fa-play")
          }
          onClick={this.togglePause}
        />
        <IconButton
          className="nextBtn fas fa-forward"
          onClick={this.nextTrack}
        />
        <IconButton
          className={
            "likeBtn " +
            (this.props.likes_song ? "fas fa-heart" : "far fa-heart")
          }
          onClick={this.toggleLike}
        />
      </>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  setErrorMsg: error => dispatch(setErrorMsg(error))
});

export default connect(undefined, mapDispatchToProps)(Player);
