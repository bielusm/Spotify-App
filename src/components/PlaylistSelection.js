import React from "react";
import {
  TableBody,
  TableHead,
  TableRow,
  TableCell,
  Table
} from "@material-ui/core";

export const PlaylistSelection = props => (
  <>
    {props.showPlaylists && (
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Your playlists</TableCell>
          </TableRow>
        </TableHead>
        <TableBody onClick={props.selectPlaylist}>
          {props.playlists.map(playlist => {
            let className = "playlist-selection__item";
            return (
              <TableRow hover selected={playlist.selected} key={playlist.name}>
                <TableCell id={playlist.id} className={className}>
                  {playlist.name}
                </TableCell>
              </TableRow>
            );
          })}
        </TableBody>
      </Table>
    )}
  </>
);

export default PlaylistSelection;
