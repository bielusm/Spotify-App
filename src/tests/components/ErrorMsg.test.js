import React from "react";
import { shallow } from "enzyme";
import { ErrorMsg } from "../../components/ErrorMsg";

test("Snapshot for no error msg", () => {
  const wrapper = shallow(<ErrorMsg errMsg={null} />);
  expect(wrapper).toMatchSnapshot();
});

test("Snapshot with error msg", () => {
  const wrapper = shallow(<ErrorMsg errMsg={"Sample error text"} />);
  expect(wrapper).toMatchSnapshot();
});
