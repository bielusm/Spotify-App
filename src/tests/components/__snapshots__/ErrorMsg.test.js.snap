// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for no error msg 1`] = `<Fragment />`;

exports[`Snapshot with error msg 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    className="my-4"
    id="errMsg"
    variant="h2"
  >
    Sample error text
  </WithStyles(ForwardRef(Typography))>
</Fragment>
`;
